builtin/providers/aws/resource_aws_codedeploy_deployment_config.go

```diff
@@ -0,0 +1,152 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/codedeploy"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeDeployDeploymentConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeDeployDeploymentConfigCreate,
+		Read:   resourceAwsCodeDeployDeploymentConfigRead,
+		Delete: resourceAwsCodeDeployDeploymentConfigDelete,
+
+		Schema: map[string]*schema.Schema{
+			"deployment_config_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"minimum_healthy_hosts": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateMinimumHealtyHostsType,
+						},
+
+						"value": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"deployment_config_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCodeDeployDeploymentConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codedeployconn
+
+	input := &codedeploy.CreateDeploymentConfigInput{
+		DeploymentConfigName: aws.String(d.Get("deployment_config_name").(string)),
+		MinimumHealthyHosts:  expandAwsCodeDeployConfigMinimumHealthHosts(d),
+	}
+
+	_, err := conn.CreateDeploymentConfig(input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("deployment_config_name").(string))
```

Comment:
I used the name as the Terraform ID here as this is what we refer to when calling from other resources

i.e. deployment_config_name = "${aws_codedeploy_deployment_config.foo.id}"